### PR TITLE
Fix pressure calculation

### PR DESCRIPTION
**Pull request by matt-frey**

### CODE REVIEW

File: src/inversion/inversion.f90 [hunk 1]
```diff
@@ -378,9 +378,12 @@ end subroutine vorticity_tendency
         !::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::
 
         ! Compute pressure with buoyancy anomaly b' (if --enable-buoyancy and --enable-perturbation-mode)
-        ! Solves Lap(p) = R + f * zeta + b'_z with dp/dz = 0 on each boundary where
+        ! If buoyancy mode is disabled we solve:
+        ! Solves Lap(p) = R + f * zeta + b'_z with dp/dz = b' on each boundary where
         ! R = 2[J_xy(u,v) + J_yz(v,w) + J_zx(w,u)) where J_ab(f,g) = f_a * g_b - f_b * g_a
         ! is the Jacobian.
+        ! If buoyancy mode is enabled we solve:
+        ! Solves Lap(p) = R with dp/dz = 0 on each boundary where
```
Comment (daviddritschel) [inline src/inversion/inversion.f90:386]: remove where at the end of this comment

File: tests/test_vor2vel.f90 [hunk 2]
```diff
@@ -82,7 +82,7 @@ program test_vor2vel
         l = twopi
         k = two * l
 
-        k2l2 = k ** 2 - l ** 2
+        klsq = k ** 2 - l ** 2
```
Comment (daviddritschel) [inline tests/test_vor2vel.f90:85]: This is supposed to be k^2 - l^2, right, not k^2 + l^2?

**Review by daviddritschel — COMMENTED** [review]
See minor comments.  We should test this on an analytic pressure field with two consistent surface buoyancy fields.  But if it agrees with the output of balance in the same case, I believe it is correct

File: tests/test_vor2vel.f90 (hunk 2 above)
Comment (matt-frey) [reply tests/test_vor2vel.f90:85]: I can currently not remember the exact setup/equations here ... I think it is correct the way it is right now. It is just a bad naming.

**Pull request merged**